src/test/scala/examples/BundleInitSpec.scala

```diff
@@ -0,0 +1,166 @@
+// See LICENSE for license details.
+
+package examples
+
+/**
+  * This example shows how to initialize bundles with methods.
+  * Such methods need not set all fields.
+  * This illustrates a way of preserving fields values that were not specified by the method
+  */
+
+import chisel3._
+import chisel3.iotesters.PeekPokeTester
+import org.scalatest.{FreeSpec, Matchers}
+
+class MyBundle extends Bundle {
+  val x = UInt(8.W)
+  val y = UInt(8.W)
+}
+
+object MyBundle {
+  /**
+    * initialize x and y
+    * @return
+    */
+  def init1: MyBundle = {
+    val wire = Wire(new MyBundle)
+    wire.x := 8.U
+    wire.y := 9.U
+    wire
+  }
+
+  /**
+    * This init does not take an existing bundle to preserve unspecified values
+    * @return
+    */
+  def setYTo5(): MyBundle = {
+    val wire = Wire(new MyBundle)
+    wire.y := 5.U
+    wire
+  }
+  /**
+    * This init takes an existing bundle to preserve unspecified values, in this case wire.x
+    * @param current set this to preserve existing values for fields not set
+    * @return
+    */
+  def setXTo7(current: MyBundle): MyBundle = {
+    val wire = Wire(new MyBundle)
+    wire := current
+    wire.x := 7.U
+    wire
+  }
+}
```

Comment:
Would it make more sense to have the above definitions more like:
```
  def setX(current: MyBundle, x: UInt): MyBundle = {
     val wire = Wire(new MyBundle)
     wire := current
     wire.x := x
     wire
   }
```
or is that crossing too many borders?